Week3/homework/step2-3.js

```diff
@@ -30,9 +34,13 @@ console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 function repeatStringNumTimesWithDoWhile(str, num) {
   // eslint-disable-next-line prefer-const
   let result = '';
-
+  let i = 0;
   // Replace this comment and the next line with your code
   console.log(str, num, result);
+  do {
+    result = num <= 0 ? result : result + str;
```

Comment:
You are correct that you must test for `num <= 0`,  however, you need not do this for every loop iteration. It is best and more efficient to just test it once, before entering the `do-while` loop and just skip the loop altogether if the loop condition is not met.

```js
if (num > 0) {
  do {
    result += str;
    i++;
  } while (i < num);
}
```